lib/stat_tracker.rb

```diff
@@ -56,12 +53,102 @@ def percentage_visitor_wins
     (away_win?.to_f / @games.size).round(2)
   end
 
+  def total_goals_in_game
+    @games.map do|game| 
+      game.away_goals + game.home_goals
+    end
+  end
+
+  def count_of_games_by_season
+    games_by_season = {}
+    @games.each do |game|
+      if games_by_season.include?(game.season_id.to_sym)
+        games_by_season[game.season_id.to_sym] += 1 
+      else
+        games_by_season[game.season_id.to_sym] = 1 
+      end
+    games_by_season
+  end
+
+  def count_of_goals_by_season
+    goals_by_season = {}
+    @games.each do |game|
+      if goals_by_season.include?(game.season_id.to_sym)
+        goals_by_season[game.season_id.to_sym] += ([game.away_goals.to_f]+[game.away_goals.to_f])
+      elsif !goals_by_season.include?(game.season_id.to_sym)
+        goals_by_season[game.season_id.to_sym] = ([game.away_goals.to_f]+[game.home_goals.to_f])
+      end 
+    goals_by_season
+  end
+
+  def average_goals_per_game
+    (total_goals_in_game.sum / @games.count.to_f).round(2)
+  end
+
```

Comment:
I like all the helper methods you used. It makes the final methods much more readable.